app/dao/notifications_dao.py

```diff
@@ -448,28 +448,57 @@ def dao_timeout_notifications(cutoff_time, limit=100000):
 
 
 def is_delivery_slow_for_providers(
-    created_at,
+    created_within_minutes,
+    delivered_within_minutes,
     threshold,
-    delivery_time,
 ):
     """
     Returns a dict of providers and whether they are currently slow or not. eg:
     {
         'mmg': True,
         'firetext': False
     }
+
+    A provider is considered slow if more than the `threshold` of their messages
+    sent in the last `created_within_minutes` minutes took over
+    `delivered_within_minutes` minutes to be delivered
+    """
+    providers_slow_delivery_ratios = get_ratio_of_messages_delivered_slowly_per_provider(
+        created_within_minutes, delivered_within_minutes
+    )
+
+    slow_providers = {}
+    for provider, ratio in providers_slow_delivery_ratios.items():
+        slow_providers[provider] = ratio >= threshold
+        statsd_client.gauge(f"slow-delivery.{provider}.ratio", ratio)
+
+    return slow_providers
+
+
+def get_ratio_of_messages_delivered_slowly_per_provider(created_within_minutes, delivered_within_minutes):
+    """
+    Returns a dict of providers with the ratio of their messages sent in the
+    last `created_within_minutes` minutes that took over
+    `delivered_within_minutes` minutes to be delivered
+
+    {
+        'mmg': 0.4,
+        'firetext': 0.12
+    }
     """
+    created_after = datetime.utcnow() - timedelta(minutes=created_within_minutes)
+    delivered_by = timedelta(minutes=delivered_within_minutes)
```

Comment:
For those two variables, I find the original names clearer.

Reply:
Ah thanks Pea, I did wonder if these were actually improved or not and wasn't quite happy with what I picked.

Out of interest, do you happen to have any alternatives for even better names? Equally I'm happy to change these back to what they were

Reply:
I think `delivery_time` is very clear, and I wouldn't change it. `delivered_by` suggests a datetime, not a timedelta.

'created_after' is actually pretty alright. `created_since` might be even better, since we use `>=` operator, not ">"

Reply:
Thanks Pea, I've changed both to be your suggestions and squashed this change in as I don't think having distinct git history is useful for it

Reply:
Thanks a lot!